### PR TITLE
Fix bug with revoking token. Add debug log when skipping auth result

### PR DESCRIPTION
**Pull request by Krock21**
Issue: #468

Fix bug with revoking token. Add debug log when skipping auth result

### CODE REVIEW

File: yt/yt/library/auth_server/auth_cache-inl.h [hunk 1]
```diff
@@ -48,6 +53,8 @@ TFuture<TValue> TAuthCache<TKey, TValue, TContext>::Get(const TKey& key, const T
                     entry->Updating = false;
 
                     if (transientError) {
+                        YT_LOG_DEBUG("Skipping transient error while updating authentication cache (Error: %v)",
```
Comment (gritukan) [inline yt/yt/library/auth_server/auth_cache-inl.h:56]: Error is usually logged as `YT_LOG_DEBUG(error, "Some message")`

Comment (gritukan) [inline yt/yt/library/auth_server/auth_cache-inl.h:56]: I'd prefer `refreshing authentication cache entry`. Also, let's log key.

**Review by gritukan — APPROVED** [review]
LGTM besides cosmetics

**Comment by gritukan** [comment]
Btw, can we add a test for that case?

File: yt/yt/library/auth_server/config.h [hunk 2]
```diff
@@ -20,8 +20,11 @@ class TAuthCacheConfig
     : public virtual NYTree::TYsonStruct
 {
 public:
+    /// Time between last update time and entry update for correct entries.
```
Comment (gritukan) [inline yt/yt/library/auth_server/config.h:23]: /// -> //!

File: yt/yt/library/auth_server/auth_cache-inl.h [hunk 3]
```diff
@@ -47,7 +48,10 @@ TFuture<TValue> TAuthCache<TKey, TValue, TContext>::Get(const TKey& key, const T
                     auto guard = Guard(entry->Lock);
                     entry->Updating = false;
 
+                    const auto& Logger = AuthLogger;
+
                     if (transientError) {
+                        YT_LOG_DEBUG(value, "Skipping transient error while updating authentication cache entry (EntryAddress: %v, ErrorCode: %v)", entry.Get(), value.GetCode());
```
Comment (gritukan) [inline yt/yt/library/auth_server/auth_cache-inl.h:54]: I'd prefer

```
YT_LOG_DEBUG(value, "Skipping transient error while updating authentication cache entry (EntryAddress: %v, ErrorCode: %v)",
    entry.Get(),
    value.GetCode());
```

to make lines shorter.

Comment (gritukan) [inline yt/yt/library/auth_server/auth_cache-inl.h:54]: Logging value.GetCode() is not nessesary, error code will be stored in error message

Comment (gritukan) [inline yt/yt/library/auth_server/auth_cache-inl.h:54]: Why do we need to log entry address?

Comment (Krock21) [reply yt/yt/library/auth_server/auth_cache-inl.h:54]: This is not true, unfortunately :(

Here is an example of a log entry:
```
2024-04-30 18:56:48,081456      D       Auth    Skipping transient error while updating authentication cache entry (EntryAddress: 0x57fb3fc73c20, ErrorCode: 1)\nToken is missing in Cypress\n    origin          localhost (pid 1238090, thread Connection:1, fid fffef0e239922759)\n    datetime        2024-04-30T18:56:48.081355Z\n    token_hash      65605aca3c618fd53e3d0cd2b9777c91bbd3cf173d8a679b0a0687f7a599a332\n\n  Error resolving path //sys/cypress_tokens/65605aca3c618fd53e3d0cd2b9777c91bbd3cf173d8a679b0a0687f7a599a332/@user\n      code            500\n      origin          localhost (pid 1238091, thread LocalRead:3, fid fffef0da5cf0f967)\n      datetime        2024-04-30T18:56:48.079247Z\n      method          Get\n\n    Node //sys/cypress_tokens has no child with key "65605aca3c618fd53e3d0cd2b9777c91bbd3cf173d8a679b0a0687f7a599a332"\n        code            500\n        origin          localhost (pid 1238091, thread LocalRead:3, fid fffef0da5cf0f967)\n        datetime        2024-04-30T18:56:48.079166Z\n      Connection:1    fffef0e239922759        5bef5a65-8fa1846c-19d3140c-b84a69e2
```

and one more:

```
2024-05-01 08:15:00,209332      D       Auth    Skipping transient error while updating authentication cache entry (EntryAddress: 0x1500bfc42d20, ErrorCode: 1)\nBlackbox call failed\n    origin          localhost (pid 1355810, thread RpcLight, fid fffef0e0315c65d3)\n    datetime        2024-05-01T08:15:00.209070Z\n    call_id         1e78e781-a8a6a66f-f67cb512-33eb19b0\n\n  Blackbox call attempt 1 failed\n      origin          localhost (pid 1355810, thread RpcLight, fid fffef0e0315c65d3)\n      datetime        2024-05-01T08:14:55.209009Z\n      attempt         1\n      call_id         1e78e781-a8a6a66f-f67cb512-33eb19b0\n\n    Blackbox call failed\n        origin          localhost (pid 1355810, thread RpcLight, fid fffef0e0315c65d3)\n        datetime        2024-05-01T08:14:55.208742Z\n        call_id         1e78e781-a8a6a66f-f67cb512-33eb19b0\n\n      Operation timed out\n          code            3\n          origin          localhost (pid 1355810, thread DelayedExecutor, fid fffef0dfeeb1d88a)\n          datetime        2024-05-01T08:14:55.208604Z\n          timeout         10000\n\n  Blackbox call attempt 2 failed\n      origin          localhost (pid 1355810, thread RpcLight, fid fffef0e0315c65d3)\n      datetime        2024-05-01T08:15:00.209044Z\n      attempt         2\n      call_id         1e78e781-a8a6a66f-f67cb512-33eb19b0\n\n    Blackbox call failed\n        origin          localhost (pid 1355810, thread RpcLight, fid fffef0e0315c65d3)\n        datetime        2024-05-01T08:15:00.208759Z\n        call_id         1e78e781-a8a6a66f-f67cb512-33eb19b0\n\n      Operation timed out\n          code            3\n          origin          localhost (pid 1355810, thread DelayedExecutor, fid fffef0dfcd904d85)\n          datetime        2024-05-01T08:15:00.208647Z\n          timeout         3999\n   RpcLight        fffef0e0315c65d3           f431bd0b-a4137c07-4b92d44f-d24e1be3
```

code was added while I was trying to debug why is it skipping my error

Comment (Krock21) [reply yt/yt/library/auth_server/auth_cache-inl.h:54]: To be able to match differrent log entries(are they about the same entry?) and debug cases similar to "why cache with TTL of 5 seconds returns stale value for 30 minutes"
The best way would be to log key, but they are sensitive now ('ytct-abcd-efghji')
So as a tradeoff I decided to log entry address

Comment (Krock21) [reply yt/yt/library/auth_server/auth_cache-inl.h]: Turns out it doesn't add generic error code (1) to attributes, but adds all others

File: yt/yt/library/auth_server/auth_cache-inl.h [hunk 4]
```diff
@@ -5,7 +5,9 @@
 #endif
 
 #include "config.h"
+#include "private.h"
 
+#include <yt/yt/core/crypto/crypto.h>
```
Comment (gritukan) [inline yt/yt/library/auth_server/auth_cache-inl.h:10]: Please add a blank line between two directories

**Review by gritukan — APPROVED** [review]

**Comment by Krock21** [comment]
The pull request is ready to be imported
Please cherry-pick it to 23.2 and 24.1 too

**Comment by chizhonkova** [comment]
Shipped!

**Pull request closed without merging**